src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/ChunkContextToolbar.java

```diff
@@ -168,10 +186,96 @@ public void onBrowserEvent(Event event)
       });
    }
    
+   private MenuItem createMenuItemForType(final AppCommand command, final String chunkType)
+   {
+      SafeHtml menuHTML = new SafeHtmlBuilder().appendHtmlConstant(
+         command.getMenuHTML(false)).toSafeHtml();
+      
+      MenuItem menuItem = new MenuItem(
+         menuHTML,
+         new Command()
+         {
+            public void execute()
+            {
+               host_.switchChunk(chunkType);
+            }
+         });
+      
+      return menuItem;
+   }
+
+   private void initChangeChunkEngine(String engine)
+   {
+      chunkTypeLabel_.setText(engine);
+      DOM.sinkEvents(chunkTypePanel_.getElement(), Event.ONCLICK);
+      DOM.setEventListener(chunkTypePanel_.getElement(), new EventListener()
+      {
+         @Override
+         public void onBrowserEvent(Event event)
+         {
+            if (DOM.eventGetType(event) == Event.ONCLICK)
+            {
+               Commands commands = RStudioGinjector.INSTANCE.getCommands();
+               String engineLabel = chunkTypeLabel_.getText();
+               
+               final ToolbarPopupMenu switchChunksMenu = new ToolbarPopupMenu();
+               
+               if (engineLabel != "r") {
+                  switchChunksMenu.addItem(createMenuItemForType(
+                        commands.switchToChunkR(), "r"));
+                  switchChunksMenu.addSeparator();
+               }
+               
+               if (!BrowseCap.isWindows() && engineLabel != "bash") {
```

Comment:
Should be `isWindowsDesktop()` -- otherwise users on e.g. Windows using Chrome to access an RStudio server on Linux won't see this option.